client/src/__tests__/Event/Select/EventSelect.test.tsx

```diff
@@ -0,0 +1,141 @@
+    });
+
+    it("can create and select an event type", async () => {
+        await setupEventSelectPromise();
+        td.when(mockGetEventTypeContains(td.matchers.anything())).thenResolve(
+            makePage() as PageDTO<EventType>,
+        );
+        td.when(
+            mockCreateEventType({
+                eventTypeId: 0,
+                eventTypeName: "fourth",
+                displayName: "fourth",
+                sortOrder: 0,
+            }),
+        ).thenResolve({
+            eventTypeId: 4,
+            displayName: "fourth",
+            eventTypeName: "fourth",
+            sortOrder: 4,
+        });
+        await act(async () => {
+            await selectEvent.create(screen.getByLabelText(EVENT_TYPE_LABEL), "fourth", {
+                waitForElement: false,
+            });
+        });
+        expect(screen.getByText("fourth")).toBeInTheDocument();
+    });
+
+    it("can search for event types", async () => {
+        await setupEventSelectPromise();
+
+        td.when(mockGetEventTypeContains("fou")).thenResolve(
+            makePage({
+                items: [
+                    makeEventType({
+                        eventTypeId: 4,
+                        eventTypeName: "fourth",
+                        displayName: "fourth",
+                        sortOrder: 4,
+                    }),
+                ],
+            }) as PageDTO<EventType>,
+        );
+        fireEvent.change(screen.getByLabelText(EVENT_TYPE_LABEL), {
+            target: { value: "fou" },
+        });
+        await selectEvent.select(screen.getByLabelText(EVENT_TYPE_LABEL), "fourth");
+        expect(screen.getByText("fourth")).toBeInTheDocument();
+    });
+
+    interface RenderCreateEventProps {
+        history?: MemoryHistory;
+        eventTypesPromise?: Promise<PageDTO<unknown>>;
+        orgNamesPromise?: Promise<PageDTO<unknown>>;
+        event?: TsrEvent;
+    }
+
+    const renderCreateEvent = async ({
+        history = createMemoryHistory(),
+        eventTypesPromise = Promise.resolve(makePage()),
+        orgNamesPromise = Promise.resolve(makePage()),
+        event,
+    }: RenderCreateEventProps): Promise<RenderResult> => {
+        history.push(event ? `/editEvent/${event.eventId}` : "/createEvent");
+
+        if (event) {
+            td.when(mockGetEventById(td.matchers.anything())).thenResolve(event);
+        }
+        td.when(mockGetEventTypeContains("")).thenDo(() => Promise.resolve(eventTypesPromise));
+        td.when(mockGetOrganizationContains("")).thenDo(() => Promise.resolve(orgNamesPromise));
+        td.when(mockCurrentDate()).thenReturn(new Date(1607760000000)); // 12/12/2020
+        td.when(mockDatePlusYears(10)).thenReturn(1923292800000); // 12/12/2030
+
+        const path = event ? "/editEvent/:eventId" : "/createEvent";
+        const result = render(
+            <Router history={history}>
+                <Route path={path}>
+                    <CreateEvent />
```

Comment:
Look at OrgSelect test's `OrgSelectTestComponent` function for setup. You do not need to render the entire CreateEvent just to test the EventSelect. This will allow you to eliminate most of the setup you currently have in terms of mocking out the OrgSelect and dates.